lib/utils/selector.js

```diff
@@ -0,0 +1,582 @@
+ * @param {string} selector CSS selector
+ * @param {RuleContext} context - The rule context.
+ * @returns {VElementSelector}
+ */
+function parseSelector(selector, context) {
+  let astSelector
+  try {
+    astSelector = parser().astSync(selector)
+  } catch (e) {
+    context.report({
+      loc: { line: 0, column: 0 },
+      message: `Cannot parse selector: ${selector}.`
+    })
+    return {
+      test: () => false
+    }
+  }
+
+  try {
+    const test = selectorsToVElementMatcher(astSelector.nodes)
+
+    return {
+      test(element) {
+        return test(element, null)
+      }
+    }
+  } catch (e) {
+    if (e instanceof SelectorError) {
+      context.report({
+        loc: { line: 0, column: 0 },
+        message: e.message
+      })
+      return {
+        test: () => false
+      }
+    }
+    throw e
+  }
+}
+
+class SelectorError extends Error {}
+
+/**
+ * @typedef {(element: VElement, subject: VElement | null )=>boolean} VElementMatcher
+ * @typedef {Exclude<parser.Selector['nodes'][number], {type:'comment'|'root'}>} ChildNode
+ */
+
+/**
+ * Convert nodes to VElementMatcher
+ * @param {parser.Selector[]} selectorNodes
+ * @returns {VElementMatcher}
+ */
+function selectorsToVElementMatcher(selectorNodes) {
+  const selectors = selectorNodes.map((n) =>
+    selectorToVElementMatcher(cleanSelectorChildren(n))
+  )
+  return (element, subject) => selectors.some((sel) => sel(element, subject))
+}
+
+/**
+ * Clean and get the selector child nodes.
+ * @param {parser.Selector} selector
+ * @returns {ChildNode[]}
+ */
+function cleanSelectorChildren(selector) {
+  /** @type {ChildNode[]} */
+  const nodes = []
+  /** @type {ChildNode|null} */
+  let last = null
+  for (const node of selector.nodes) {
+    if (node.type === 'root') {
+      throw new Error('Unexpected state type=root')
+    }
+    if (node.type === 'comment') {
+      continue
+    }
+    if (
+      (last == null || last.type === 'combinator') &&
+      isDescendantCombinator(node)
+    ) {
+      // Ignore descendant combinator
+      continue
+    }
+    if (isDescendantCombinator(last) && node.type === 'combinator') {
+      // Replace combinator
+      nodes.pop()
+    }
+    nodes.push(node)
+    last = node
+  }
+  if (isDescendantCombinator(last)) {
+    nodes.pop()
+  }
+  return nodes
+
+  /**
+   * @param {parser.Node|null} node
+   * @returns {node is parser.Combinator}
+   */
+  function isDescendantCombinator(node) {
+    return Boolean(node && node.type === 'combinator' && !node.value.trim())
+  }
+}
+/**
+ * Convert Selector child nodes to VElementMatcher
+ * @param {ChildNode[]} selectorChildren
+ * @returns {VElementMatcher}
+ */
+function selectorToVElementMatcher(selectorChildren) {
+  const nodes = [...selectorChildren]
+  let node = nodes.shift()
+  /**
+   * @type {VElementMatcher | null}
+   */
+  let result = null
+  while (node) {
+    if (node.type === 'combinator') {
```

Comment:
The component-tags-order rule does not use the combinator, but it does handle the combinator so that it can be used in future changes. It has also been tested.